modules/tensor_mechanics/src/materials/ComputeThermalExpansionEigenstrainBase.C

```diff
@@ -38,7 +40,13 @@ ComputeThermalExpansionEigenstrainBaseTempl<is_ad>::ComputeThermalExpansionEigen
                            : &this->template declarePropertyDerivative<RankTwoTensor>(
                                  _eigenstrain_name, this->getVar("temperature", 0)->name())),
     _stress_free_temperature(this->coupledValue("stress_free_temperature")),
-    _temperature_prop(this->template coupledGenericValue<is_ad>("temperature"))
+    _temperature_prop(this->template coupledGenericValue<is_ad>("temperature")),
+    _calc_mean_CTE(parameters.isParamSetByUser("mean_thermal_expansion_coefficient_name")),
+    _mean_thermal_expansion_coefficient(
+        _calc_mean_CTE
+            ? &this->template declareProperty<Real>(this->template getParam<MaterialPropertyName>(
+                  "mean_thermal_expansion_coefficient_name"))
+            : nullptr)
```

Comment:
```suggestion
    _mean_thermal_expansion_coefficient(
        isParamValid("mean_thermal_expansion_coefficient_name")
            ? &this->template declareProperty<Real>(this->template getParam<MaterialPropertyName>(
                  "mean_thermal_expansion_coefficient_name"))
            : nullptr)
```

Reply:
note that you should use `isParamValid` here

Reply:
Does the `isParamValid` work in the action, too? I have a sneaking suspicion that `isParamSetByUser` is false when the action sets it instead of a user. I think `isParamValid` avoid this.

I'll make the changes and do some tests.

Reply:
Yes, `isParamValid` is the way to go.